django_declarative_apis/machinery/attributes.py

```diff
@@ -730,6 +730,60 @@ def _run_task(self, owner_instance):
             )
 
 
+class DeferrableGenericEndpointTask(DeferrableEndpointTask):
+    # very similar to DeferrableEndpointTask, but doesn't assume that the resource is a Django model instance
+
+    def __init__(
+        self,
+        task_args_packer=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        assert (
```

Comment:
My thinking is that because any incorrect usage of this decorator will reveal itself at compile time, was can use an `assert` rather than raising a `TypeError` and get the same protection because any issue will be reliably detected in unit tests.  Using the `assert` allows the check to be optimized out when running in production